fdbserver/Status.actor.cpp

```diff
@@ -1919,74 +1923,69 @@ static Future<std::vector<TraceEventFields>> getServerBusiestWriteTags(
 }
 
 ACTOR
-static Future<std::vector<Optional<StorageMetadataType>>> getServerMetadata(std::vector<StorageServerInterface> servers,
-                                                                            Database cx,
-                                                                            bool use_system_priority) {
+static Future<std::vector<StorageServerStatusInfo>> readStorageInterfaceAndMetadata(Database cx,
+                                                                                    bool use_system_priority) {
 	state KeyBackedObjectMap<UID, StorageMetadataType, decltype(IncludeVersion())> metadataMap(serverMetadataKeys.begin,
 	                                                                                           IncludeVersion());
-	state std::vector<Optional<StorageMetadataType>> res(servers.size());
 	state Reference<ReadYourWritesTransaction> tr = makeReference<ReadYourWritesTransaction>(cx);
-
+	state std::vector<StorageServerStatusInfo> servers;
 	loop {
 		try {
+			servers.clear();
 			tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
 			tr->setOption(FDBTransactionOptions::LOCK_AWARE);
 			if (use_system_priority) {
 				tr->setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
 			}
-			state int i = 0;
-			for (i = 0; i < servers.size(); ++i) {
-				Optional<StorageMetadataType> metadata = wait(metadataMap.get(tr, servers[i].id(), Snapshot::True));
+			state RangeResult serverList = wait(tr->getRange(serverListKeys, CLIENT_KNOBS->TOO_MANY));
+			ASSERT(!serverList.more && serverList.size() < CLIENT_KNOBS->TOO_MANY);
+
+			servers.reserve(serverList.size());
+			for (int i = 0; i < serverList.size(); i++) {
+				servers.push_back(StorageServerStatusInfo(decodeServerListValue(serverList[i].value)));
+			}
+			state std::vector<Future<Void>> futures(servers.size());
+			for (int i = 0; i < servers.size(); ++i) {
+				auto& info = servers[i];
+				futures[i] = fmap(
```

Comment:
There is actually a generic ACTOR you can use here called `Future<Void> store(T &dst, Future<T> sourceFuture)`

Reply:
I think this will work:
```
state std::vector<Future<Void>> futures(servers.size());
for (int i = 0; i < servers.size(); ++i) {
	futures[i] = store(servers[i].info, metadataMap.get(tr, servers[i].id()));
}
```